cmd/promtool/tsdb.go

```diff
@@ -184,24 +91,24 @@ func (b *writeBenchmark) run() error {
 	if err != nil {
 		return err
 	}
+	st.DisableCompactions()
```

Comment:
I am not sure about this. When I tested `tsdb bench write`, I saw one compaction failure in the `stopStorage` stage. I think this is caused by https://github.com/prometheus/prometheus/pull/6088/files#diff-587d0d6cc9fcdcf5edcc7dbe2c79cf96R134-R137, we close the tsdb here so it causes a race for auto compaction. 

So I disable the auto compaction to prevent the race, does it make sense?

```
>> running benchmark, writing result to ./benchout
level=info ts=2020-07-17T04:04:42.711295339Z caller=head.go:641 msg="Replaying on-disk memory mappable chunks if any"
level=info ts=2020-07-17T04:04:42.71132615Z caller=head.go:655 msg="On-disk memory mappable chunks replay completed" duration=1.697µs
level=info ts=2020-07-17T04:04:42.7113348Z caller=head.go:661 msg="Replaying WAL, this may take a while"
level=info ts=2020-07-17T04:04:42.711515635Z caller=head.go:713 msg="WAL segment loaded" segment=0 maxSegment=0
level=info ts=2020-07-17T04:04:42.71152925Z caller=head.go:716 msg="WAL replay completed" checkpoint_replay_duration=15.97µs wal_replay_duration=171.071µs total_replay_duration=204.467µs
>> start stage=readData
>> completed stage=readData duration=9.010239ms
>> start stage=ingestScrapes
level=info ts=2020-07-17T04:04:42.911077592Z caller=compact.go:495 msg="write block" mint=30000 maxt=7200000 ulid=01EDDF46P7MCDN4Q1EJ5X3ZWPG duration=87.424018ms
level=info ts=2020-07-17T04:04:42.916771849Z caller=head.go:804 msg="Head GC completed" duration=1.796868ms
level=info ts=2020-07-17T04:04:42.980048205Z caller=compact.go:495 msg="write block" mint=7200000 maxt=14400000 ulid=01EDDF46SC2ZG9GQQYYKWZ2VQY duration=55.081893ms
level=info ts=2020-07-17T04:04:42.985088429Z caller=head.go:804 msg="Head GC completed" duration=1.785562ms
level=info ts=2020-07-17T04:04:43.081263835Z caller=compact.go:495 msg="write block" mint=14400000 maxt=21600000 ulid=01EDDF46VFGNKW589YA66EYBTP duration=89.558609ms
level=info ts=2020-07-17T04:04:43.105916664Z caller=head.go:804 msg="Head GC completed" duration=8.574277ms
level=info ts=2020-07-17T04:04:43.192507556Z caller=compact.go:495 msg="write block" mint=21600000 maxt=28800000 ulid=01EDDF46ZJQ8CH6ZW4227QWP3X duration=70.337554ms
level=info ts=2020-07-17T04:04:43.197806232Z caller=head.go:804 msg="Head GC completed" duration=1.697093ms
level=info ts=2020-07-17T04:04:43.201031161Z caller=checkpoint.go:96 msg="Creating checkpoint" from_segment=0 to_segment=1 mint=28800000
level=info ts=2020-07-17T04:04:43.2503322Z caller=head.go:884 msg="WAL checkpoint complete" first=0 last=1 duration=49.352097ms
level=info ts=2020-07-17T04:04:43.308328754Z caller=compact.go:495 msg="write block" mint=28800000 maxt=36000000 ulid=01EDDF473MXBJW38VEQFY7VMTC duration=55.554547ms
level=info ts=2020-07-17T04:04:43.313698535Z caller=head.go:804 msg="Head GC completed" duration=1.967459ms
level=info ts=2020-07-17T04:04:43.369380643Z caller=compact.go:495 msg="write block" mint=36000000 maxt=43200000 ulid=01EDDF475S6XFJBPWQ58VQC3FF duration=47.424709ms
level=info ts=2020-07-17T04:04:43.375949209Z caller=head.go:804 msg="Head GC completed" duration=1.867668ms
level=info ts=2020-07-17T04:04:43.380923622Z caller=checkpoint.go:96 msg="Creating checkpoint" from_segment=2 to_segment=3 mint=43200000
level=info ts=2020-07-17T04:04:43.410941297Z caller=head.go:884 msg="WAL checkpoint complete" first=2 last=3 duration=30.100943ms
level=info ts=2020-07-17T04:04:43.46156506Z caller=compact.go:495 msg="write block" mint=43200000 maxt=50400000 ulid=01EDDF478PSTASAHX6PSFVY49S duration=47.192342ms
level=info ts=2020-07-17T04:04:43.467009835Z caller=head.go:804 msg="Head GC completed" duration=1.715044ms
level=info ts=2020-07-17T04:04:43.517852121Z caller=compact.go:495 msg="write block" mint=50400000 maxt=57600000 ulid=01EDDF47AHZAGK0DC74Z0F89TK duration=43.879272ms
level=info ts=2020-07-17T04:04:43.522848234Z caller=head.go:804 msg="Head GC completed" duration=1.889164ms
level=info ts=2020-07-17T04:04:43.526273196Z caller=checkpoint.go:96 msg="Creating checkpoint" from_segment=4 to_segment=5 mint=57600000
level=info ts=2020-07-17T04:04:43.575024041Z caller=head.go:884 msg="WAL checkpoint complete" first=4 last=5 duration=48.821878ms
ingestion completed
>> completed stage=ingestScrapes duration=860.843044ms
 > total samples: 3000000
 > samples/sec: 3.484920271389978e+06
>> start stage=stopStorage
level=error ts=2020-07-17T04:04:43.583874046Z caller=db.go:685 msg="compaction failed" err="persist head block: 2 errors: write compaction: add symbol: context canceled; context canceled"
>> completed stage=stopStorage duration=161.018512ms
```

Reply:
Let's have some issue to track this. I think this is overall fine for this piece of work, but we should investigate anyway :thinking: